pandas/tslib.pyx

```diff
@@ -1388,6 +1388,26 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
 
     return convert_to_tsobject(ts, tz, unit)
 
+def _parse_iso8601(object ts):
+    '''
+    Parse string into Timestamp using iso8601 parser. Used ONLY for
```

Comment:
It's not completely necessary, but only way to test certain invalid raise in the iso parsing, since `Timestamp(...)` falls back to to dateutil.  For example YYYYMM is not valid iso, but dateutil will parse if it matches YYMMDD -

```
In [1]: pd.Timestamp('200101')                
Out[1]: Timestamp('2020-01-01 00:00:00')      
```